### DISCUSSION TITLE
How to deploy configuration files?

### DISCUSSION THREAD

**Discussion by tlull** [Keyfactor/ejbca-ce#580] [Q&A] [answered]
Hello,
I've just upgraded to version 8.1.0.2 CE. 
Now I want to change the database JDBC URL.
If I remember right, in previous versions the was the the ant target "deploy" which deploys the application and
the data specified in the configuration files to standalone.xml.
But this target is now deprecated.
In my opinion "ant runinstall" is no replacement because it generates new keystores.
What is the right way just to deploy the contents of the configuration files?

Regards,
   Torsten

**Comment by primetomas** [comment, marked as answer]
The database URL is configured directly in the application server. See: https://doc.primekey.com/ejbca/ejbca-installation/application-servers

if you need to change a property file in ejbca.ear, a simple "ant build deployear" copies a new ejbca.ear file for the application server deployment directory. EJBCA does not do application server configuration any longer, it's better done in the application server.

**Reply by tlull** [reply]
I have one additonal question. The configuration file conf/database.properties still exists. When building the ear the contained database JDBC URL is printed out. Does that has any meaning? Is the JDBC URL from database.properties used anywhere within the built ear? Or is the whole connection configuration done within Wildfly?

Kind regards,
   Torsten

**Reply by primetomas** [reply]
This does not have any meaning. The url in database properties can be ignored for the ear file and does not have to be configured. 
It is used for the standalone ejbca-db-cli tool only.
The only thing you have to configure before building ejbca.ear is the database type.
This is printed in the database.properties.sample file:
```
#
# For building and installing EJBCA on an application server only database.name needs to be set. 
# The database connections are configured in the applications server.
#
```

**Reply by tlull** [reply]
Hello Tomas,
thanks for your explanation.
With the term "ejbca-db-cli tool" you mean the clientToolBox which is build with the call "ant clientToolBox". Is that right?

The configuration file database.properties.sample contains the following comment:

```
# Database connection URL.
# This is the URL used to connect to the database, used to configure a new datasource in JBoss.
```
This comment should be removed od modified if the URL is not used for configuring the application server. This is misleading.

Kind regards,
   Torsten

**Reply by primetomas** [reply]
Thanks for noting, I will update the comment. 

ejbca-db-cli is this one (Enterprise only). 
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/command-line-interfaces#CommandLineInterfaces-Local_Database_CLI

**Comment by tlull** [comment]
Hello Tomas,
thanks for your answer.

Kind regards,
   Torsten
